test/tests/ver-1b/comparison_ver-1b.py

```diff
@@ -11,7 +11,7 @@
 gs = gridspec.GridSpec(1,1)
 ax = fig.add_subplot(gs[0])
 
-tmap_sol = pd.read_csv("./gold/u_vs_t.csv")
+tmap_sol = pd.read_csv("ver-1b_csv.csv")
```

Comment:
Why not use the gold file? Unless I am missing something, this is conflicting with the changes that Casey made in ver-1a to use the gold files. Using gold files helps to automatically create the documentation without running all the tests.

Reply:
I wanted to use the actual run solutions because I noticed that the gold file isn't always the same as what we claim on the documentation, and having the plots get generated from the run of an input file creates more transparency. I had mentioned this in the verification standards documentation too, I think. But your point makes sense, so I'm fine with reverting this change. In that case, we should make extra sure any PR that changes code/input files has to update the gold files and ensure they match what's in the documentation.

Reply:
Yes, absolutely. I think you should update the gold files in this PR, too.

Reply:
I already updated the gold files anyway in this case because the old ones were wrong. But yes, I'll change the Python script to use gold again.

Reply:
I think we discussed that a while ago but ideally you wouldn't have gold files but a documentation that is generated on-the-fly

Reply:
@RemDelaporteMathurin, I'm not sure I understand your point about not needing gold files. 
Gold files in TMAP8 are used to store the tests/validation/examples results. They are updated to represent the latest version of TMAP8 predictions (otherwise tests fail). So having them is an integral part of quality control. 

As what we did with ver-1a, we will use these gold file (which contain the updated results) to generate the documentation. This would ensure that figures in the documentation are automatically generated by the latest TMAP8 version, which I think is what you were referring to.

Reply:
My point is that there is no reason to hard code these files since they can be generated by TMAP8 directly. As @ChaitanyaBhave-e pointed out:

> we should make extra sure any PR that changes code/input files has to update the gold files and ensure they match what's in the documentation

Which means that as soon as someone forgets this, then the documentation is out-of-date again...

Instead, the figure should be generated when compiling the documentation, not hard-coded.

Reply:
If you take a look at the [HTM documentation](https://h-transport-materials.readthedocs.io/en/latest/user/plotting.html#plotting-a-property) we have no hard coded data for the figures and they are generated on-the-fly when compiling the documentation. We don't have to "remember" to update the data for the figures at every PR

Reply:
If a PR changes code/input files, then tests will fail is the gold file is not updated as well, and the PR will not get merged until the gold files are updated (or if TMAP8 behavior was supposed to remain the same, then the change is corrected).

The gold files are therefore always reflecting the latest version of TMAP8, and tying the documentation to them ensures that the documentation is always reflecting the latest version of TMAP8. 

The documentation is here generated on the fly without having to run any of the tests.

Reply:
To be very clear: gold files are not hard-coded result files used only to create the documentation. 
Gold files are the reference files that define the gold standards for up-to-date TMAP8's predictions. 

When we perform tests in a PR (see at the bottom of the page), TMAP8 is compiled and run and its predictions are compared against the gold files. If a change affects TMAP8's behavior and the results do not match the gold file, then the tests will tell us that that specific test failed. To pass the test, the developer will have to update the old file, or fix the unexpected change in behavior. 
That's why gold files are the very best source of data to build an up-to-date documentation.

Running a fresh version of TMAP8 (without local changes) would therefore produce data that matches exactly the gold files (otherwise tests would have failed). Building the documentation on the gold files alleviate the need to run these tests (which can take a significant amount of time in the future as the number of tests/validation/example cases increases).

Reply:
Right so maybe my confusion comes from the fact that I thought TMAP8 transitioned to testing produced results against analytical data not previously (gold) produced data.

I understand what you mean.
I still believe that to be more rigorous the CI should compare the results against analytical solution produced on the fly (from the python functions) instead of comparing against previous versions of the code

Reply:
We do both.

The gold files are the reference files that define the gold standards for up-to-date TMAP8's predictions. They help us compare against previous versions of the code. This way, we are made aware of any changes that would affect TMAP8 predictions. 

Meanwhile, the documentation uses the python scripts to read the gold files AND calculate the analytical expression to generate a figure that contains both and keep an up to date documentation. So TMAP8 test results are compared both against the analytical solution (when it exists) AND the previous version of the code. 
This is all done properly for the ver-1a case. 

When no analytical solution is available, then we can use benchmarking results or experimental data instead (or sometimes there's nothing to compare against). In any case, new capabilities are always tested against TMAP8's previous version (via the gold files). This ensures robustness and stability.